rfcs/20201014-VTune-ITT-tagging/README.md

```diff
@@ -0,0 +1,142 @@
+# Introducing VTune ITT tagging for primitive execution 
+
+## Introduction
+
+For deep learning programmers, function hotspots might not help a lot for 
+analyzing their deep learning workloads. Primitives level or deep learning 
+operations level hotspots mean more to deep learning developers.  
+In order to provide primitives level or DL operations level information to
+DL developers, we introduce ITT tagging for oneDNN primitive execution.
+
+With this ITT tagging for oneDNN primitive execution, DL developers can understand
+more information from VTune for below items.
+
+### 1. get the primitives timeline chart from VTune, and identify any protential performance issue.  
+For below diagrams, you can see both convolution and reorder primitives are tagged among 
+master thread and worker threads in the timeline chart.  
+We also tagged the PyTorch operations, so users can easily map one PyTorch convolution ops to one 
+convolution primitive and one reorder primitive.   
+For optimizing performance further, users can try to avoid the extra reorder primitive identified via VTune timeline.  
+![](images/timeline.JPG "oneDNN primitives timeline chart")
+
+### 2. get platform information such as L1/L2 cache miss or level of FP vectorization on primitive level   
+For below diagram, users can group profiling results by Task Types, and then VTune will group information by
+oneDNN primitives tagged as different Task Types.   
+Therefore, users can see platform level detail information like FP vectorization on primitives level.  
+Users can understand performance issues among each oneDNN primitives.  
+![](images/group.JPG "group by oneDNN primitives")
+
+### 3. map primitive with related computation kernels.  
+By tagging primitives, users can also map primitive with computation kernels.  
+For example, users can understand one inner primitive contains several gemm_nocopy_f32 kernels.  
+![](images/ops_mapping.JPG "mapping between oneDNN primitives and kernels")
+
+## Proposal
+
+Add VTune ITT tasks for all primitives.
+Users can see profiling results grouped by primitives. 
+primitive grouping is more useful than function grouping to DL developers.
+### Implementation
+To enable VTune ITT tagging, we includes ittnotify related source files in projects https://github.com/intel/ittapi under src/common/ittnofiy folder.  
+We also create ittnotify.cpp and ittnotify.hpp under /src/common folder to use ITT functions from copied ITT source files.
+We introduce a new namespace itt under dnnl namespace.
+In the same sapce we have 4 DNNL_API for using ITT tagging.
+
+| functions  | Note |  
+|---------------------|---------------------|  
+| void DNNL_API primitive_task_start(primitive_kind_t kind) |  use this function to tag start of a ITT task with primitive kind info  | 
```

Comment:
for now, the primitive_task_start supports primitive tagging, and it is not a general itt_task_start. Users can indeed use ported ITT functions to tag their functions in DNNL. 
Therefore, it might be better to still use old primitive_task_start because it is only for primitive tagging.

Reply:
Makes sense, thank you!